apps/prairielearn/src/lib/editors.js

```diff
@@ -166,6 +193,7 @@ class Editor {
                 );
               } catch (err) {
                 await cleanAndResetRepository(this.course, gitEnv, job);
+                await syncCourseFromDisk(this.course, startGitHash, job);
```

Comment:
Should we also `syncCourseFromDisk` if pushing fails? It seems like the new strategy is to always re-sync on failure.